src/lib/internal/helpers/focus.ts

```diff
@@ -0,0 +1,39 @@
+import {
+	handleRovingFocus,
+	isFunction,
+	isHTMLElement,
+	sleep,
+} from '$lib/internal/helpers/index.js';
+
+export type FocusTarget = string | HTMLElement | SVGElement | null;
+export type FocusProp = FocusTarget | ((defaultEl?: HTMLElement) => FocusTarget);
+
+type HandleFocusArgs = {
+	prop?: FocusProp;
+	defaultEl: HTMLElement;
+	roving?: boolean;
```

Comment:
Hmm, this is a bit confusing to me. Why do we deal with roving here as well? And why only with defaultEls?

Reply:
Roving focus is for the menus, and we only modify the tab index's for elements we own/control, which is why it's only for the default el :)